packages/fx-core/src/common/tools.ts

```diff
@@ -427,6 +430,41 @@ export function isAADEnabled(solutionSettings: AzureSolutionSettings): boolean {
   }
 }
 
+export function canAddSso(projectSettings: ProjectSettings): boolean {
+  if (!isAadManifestEnabled()) {
+    return false;
+  }
+
+  const solutionSettings = projectSettings.solutionSettings as AzureSolutionSettings;
+  if (!(solutionSettings.hostType === HostTypeOptionAzure.id)) {
+    return false;
+  }
+
+  if (solutionSettings.capabilities.includes(BotOptionItem.id)) {
+    const botHostType = projectSettings.pluginSettings?.[ResourcePlugins.Bot]?.[BotHostTypeName];
+    if (botHostType === BotHostTypes.AzureFunctions) {
+      return false;
+    }
+  }
+
+  const containTabSsoItem = solutionSettings.capabilities.includes(TabSsoItem.id);
+  const containBotSsoItem = solutionSettings.capabilities.includes(BotSsoItem.id);
+  const containTab = solutionSettings.capabilities.includes(TabOptionItem.id);
+  const containBot = solutionSettings.capabilities.includes(BotOptionItem.id);
+  const containAadPlugin = solutionSettings.activeResourcePlugins.includes(PluginNames.AAD);
```

Comment:
`containAadPlugin` variable is not used?

Reply:
Fixed. Thx!